packages/perseus/src/util/parse-perseus-json/exhaustive-test-tool/index.ts

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env -S node -r @swc-node/register
+
+// This is a NodeJS script that tests the Perseus JSON parsing code against a
+// content data dump.
+//
+// USAGE:
+// First, download the content data dump:
+//
+//     mkdir ~/Desktop/content
+//     gsutil -m cp -r gs://content-property.khanacademy.org/{Article,Exercise}.{Translated,}PerseusContent ~/Desktop/content
+//
+// Then, run the test tool over the content like so:
+//
+//     find ~/Desktop/content/*/* -type d | xargs -n1 packages/perseus/src/util/parse-perseus-json/exhaustive-test-tool/index.ts
+
+import {createHash} from "crypto";
+import * as fs from "fs/promises";
+import {join} from "path";
+
+import {ErrorTrackingParseContext} from "../error-tracking-parse-context";
+import {formatPath} from "../object-path";
+import {parsePerseusItem} from "../perseus-parsers/perseus-item";
+import {isSuccess} from "../result";
+
+import type {Mismatch} from "../parser-types";
+import type {Dirent} from "fs";
+
+async function main() {
+    // process.argv.slice(2) is a common NodeJS idiom. The first two args are
+    // `node` and the path to this script, so we can get rid of them.
+    const realArgs = process.argv.slice(2);
+    const {inputDir} = parseCommandLineArguments(realArgs);
+    console.log("testing files in " + inputDir);
+
+    let numFiles = 0;
+    for await (const dirent of listFilesRecursively(inputDir)) {
+        await testFile(getPathOfDirent(dirent));
+        numFiles++;
+    }
+
+    console.log("tested " + numFiles + " files");
+}
+
+function parseCommandLineArguments(args: string[]) {
+    return {inputDir: args[0]};
+}
+
+async function testFile(path: string) {
+    if (!path.endsWith(".json")) {
+        return;
+    }
+    const json = await fs.readFile(path, "utf-8");
+    let assessmentItems: null | unknown[] = null;
+    try {
+        assessmentItems = JSON.parse(json);
+    } catch {
+        console.warn("Failed to parse JSON file: " + path);
+        return;
+    }
+    if (!Array.isArray(assessmentItems)) {
+        return;
+    }
+
+    for (const rawItem of assessmentItems.map(getAssessmentItemData)) {
+        for (const mismatch of getMismatches(rawItem)) {
+            const desc = describeMismatch(mismatch);
+            const hash = sha256(desc);
+            await fs.mkdir("/tmp/test-results/" + hash, {recursive: true});
```

Comment:
It's pretty awkward that this output directory is hardcoded. I should make it an argument to the script.